services/src/main/java/org/keycloak/protocol/oid4vc/issuance/signing/JwtSigningService.java

```diff
@@ -29,14 +42,71 @@
  */
 public class JwtSigningService extends SigningService<String> {
 
-    public JwtSigningService(KeycloakSession keycloakSession, String keyId, String algorithmType) {
+    private static final Logger LOGGER = Logger.getLogger(JwtSigningService.class);
+
+    private static final String ID_TEMPLATE = "urn:uuid:%s";
+    private static final String TOKEN_TYPE = "JWT";
+    private static final String VC_CLAIM_KEY = "vc";
+    private static final String ID_CLAIM_KEY = "id";
+
+
+    private final SignatureSignerContext signatureSignerContext;
+    private final TimeProvider timeProvider;
+    protected final String issuerDid;
+
+    public JwtSigningService(KeycloakSession keycloakSession, String keyId, String algorithmType, String issuerDid, TimeProvider timeProvider) {
         super(keycloakSession, keyId, algorithmType);
+        this.issuerDid = issuerDid;
+        this.timeProvider = timeProvider;
+        KeyWrapper signingKey = getKey(keyId, algorithmType);
+        if (signingKey == null) {
+            throw new SigningServiceException(String.format("No key for id %s and algorithm %s available.", keyId, algorithmType));
+        }
+        signatureSignerContext = switch (algorithmType) {
```

Comment:
It is just an idea, to avoid this hard-cording, how about using the mechanism implemented in  `DefaultTokenManager`?

```
        SignatureProvider signatureProvider = session.getProvider(SignatureProvider.class, signatureAlgorithm);
        SignatureSignerContext signer = signatureProvider.signer();

```
But some mechanism to set the `signingKey` to the `SignerContext` is needed.

Reply:
Changed it to use the mechanism proposed by @tnorimat  - see [62eeda2](https://github.com/keycloak/keycloak/pull/26484/commits/62eeda2ee8d965a318bfbc9f71f60fa54b70078e)

Reply:
@wistefan Thank you. I confirmed the revised codes.